sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClientBuilder.java

```diff
@@ -33,27 +30,61 @@
 
-    private final List<HttpPipelinePolicy> additionalPolicies = new ArrayList<>();
+    // This is the name of the properties file in this repo that contains the default properties
+    private static final String DIGITAL_TWINS_PROPERTIES = "azure-digital-twins.properties";
+
+    // These are the keys to the above properties file that define the sdk's name and version for use in the user agent string
+    private static final String SDK_NAME = "name";
+    private static final String SDK_VERSION = "version";
+
+    private final List<HttpPipelinePolicy> additionalPolicies;
+
     // mandatory
     private String endpoint;
     private TokenCredential tokenCredential;
+
     // optional/have default values
     private DigitalTwinsServiceVersion serviceVersion;
     private HttpPipeline httpPipeline;
     private HttpClient httpClient;
-    private HttpLogOptions logOptions;
-    private RetryPolicy retryPolicy;
+    private HttpLogOptions httpLogOptions;
+    private HttpPipelinePolicy retryPolicy;
+
+    // Right now, Azure Digital Twins does not send a retry-after header on its throttling messages. If it adds support later, then
+    // these values should match the header name (for instance, "x-ms-retry-after-ms" or "Retry-After") and the time unit
+    // of the header's value. These null values are equivalent to just constructing "new RetryPolicy()"
+    private static final String retryAfterHeader = null;
+    private static final ChronoUnit retryAfterTimeUnit = null;
+    private static final RetryPolicy DEFAULT_RETRY_POLICY = new RetryPolicy(retryAfterHeader, retryAfterTimeUnit);
+
+    private final Map<String, String> properties;
+
+    private Configuration configuration;
+
+    public DigitalTwinsClientBuilder()
+    {
+        additionalPolicies = new ArrayList<>();
+        properties = CoreUtils.getProperties(DIGITAL_TWINS_PROPERTIES);
+        httpLogOptions = new HttpLogOptions();
+    }
 
     private static HttpPipeline buildPipeline(TokenCredential tokenCredential, String endpoint,
-                                              HttpLogOptions logOptions, HttpClient httpClient,
-                                              List<HttpPipelinePolicy> additionalPolicies, RetryPolicy retryPolicy) {
+                                              HttpLogOptions httpLogOptions, HttpClient httpClient,
+                                              List<HttpPipelinePolicy> additionalPolicies, HttpPipelinePolicy retryPolicy,
+                                              Configuration configuration, Map<String, String> properties) {
         // Closest to API goes first, closest to wire goes last.
         List<HttpPipelinePolicy> policies = new ArrayList<>();
 
+        String clientName = properties.getOrDefault(SDK_NAME, "UnknownName");
+        String clientVersion = properties.getOrDefault(SDK_VERSION, "UnknownVersion");
```

Comment:
These values come from azure-digital-twins.properties file that is added in this PR